src/Common/src/System/Net/SafeCloseSocket.Windows.cs

```diff
@@ -33,52 +33,56 @@ public ThreadPoolBoundHandle IOCPBoundHandle
         // Binds the Socket Win32 Handle to the ThreadPool's CompletionPort.
         public ThreadPoolBoundHandle GetOrAllocateThreadPoolBoundHandle(bool trySkipCompletionPortOnSuccess)
         {
+            // Check to see if the socket native _handle is already
+            // bound to the ThreadPool's completion port.
+            if (_released || _iocpBoundHandle == null)
```

Comment:
Could we simplify this to a single test?  Is _iocpBoundHandle always null when _released == true?  Seems like it should be...